### PR TITLE
Feature/add run profiling for thunder

### PR DESCRIPTION
**Pull request by dgergel**
This PR adds a section to the test profiling and scaling suite to enable testing on Thunder.

### CODE REVIEW

**Comment by jhamman** [comment]
this looks good to me. Once you have some results, post them here and we can merge this.

File: tests/run_profiling.py [hunk 1]
```diff
@@ -111,6 +111,39 @@ def log2_range(m):
...
+                                    dict(select=3, mpiprocs=36),
+                                    dict(select=4, mpiprocs=36),
+                                    dict(select=5, mpiprocs=36),
+                                    dict(select=6, mpiprocs=36),
+                                    dict(select=8, mpiprocs=36),
+                                    dict(select=10, mpiprocs=36),
+                                    dict(select=12, mpiprocs=36),
+				                    dict(select=24, mpiprocs=36),
+				                    dict(select=48, mpiprocs=36),
+				                    dict(select=96, mpiprocs=36)],
```
Comment (bartnijssen) [inline tests/run_profiling.py]: fix indentation

**Comment by bartnijssen** [comment]
@dgergel : As @jhamman requested - please show some output before we merge. Also merge latest `develop` into this branch. If we can complete this that would be great.

**Comment by dgergel** [comment]
Here are some figures showing profiling: 

eb_fs_noce_3nodes

eb_nofs_noce_3nodes

wb_fs_noce_3nodes

wb_fs_noce_50nodes

wb_nofs_noce_3nodes

These were run for one year without writing output on Thunder. 

I have not been able to complete two runs -  "Full Energy, Frozen Soils, no Close Energy, 50 Nodes" and "Full Energy, Frozen Soils, Close Energy, 50 Nodes" -  because they're hanging, e.g. not completing within 35 hours (for only 1 year). I'm not sure yet why this is - originally I thought that it was because they were writing to the same files at the same time, but eliminating output didn't solve the problem.

**Comment by dgergel** [comment]
Posting updated figures, these are run over the RASM domain (rather than the global domain as above), thus they have ~26000 grid cells rather than ~61000. 

eb_fs_ce_50nodes

eb_fs_noce_3nodes

eb_fs_noce_50nodes

eb_nofs_noce_3nodes

wb_fs_noce_3nodes

wb_fs_noce_50nodes

wb_nofs_noce_3nodes

This should be ready to merge (it's failing because of the CESM driver changes implemented in #692).

**Pull request merged**